src/app/events/[slug]/components/ScheduleSection/Tabs.tsx

```diff
@@ -18,29 +20,39 @@ type TabsProps = {
 
 const { screens } = theme
 
+const QUERY_KEY = 'tab'
+const DEFAULT_TAB_INDEX = 0
+
 export function Tabs({ schedule }: TabsProps) {
   const sortedDays = filterAndSortScheduleDays(schedule)
 
+  const [activeTabIndex, setActiveTabIndex] = useQueryState(
+    QUERY_KEY,
+    parseAsInteger.withDefault(DEFAULT_TAB_INDEX).withOptions({
+      clearOnDefault: false,
+    }),
+  )
+
   const tabGroupRef = useRef<HTMLDivElement>(null)
   const isMounted = useIsMounted()
   const isScreenBelowLg = useMediaQuery(
     `(max-width: ${parseInt(screens.md, 10) - 1}px)`,
   )
 
-  function scrollToTabGroup() {
-    if (isMounted() && isScreenBelowLg && tabGroupRef.current) {
-      tabGroupRef.current.scrollIntoView({
-        behavior: 'smooth',
-        block: 'start',
-      })
+  function handleIndexChange(index: number) {
+    setActiveTabIndex(index)
+
+    if (isMounted() && isScreenBelowLg) {
```

Comment:
Wouldn’t it be better to let `scrollRefIntoView` handle its core functionality without adding optional chaining or guards?

```suggestion
    if (isMounted() && isScreenBelowLg && tabGroupRef.current) {
```

Reply:
Yes, I agree with that statement.

However, ref objects are always typed as a value or `null`.

```ts
interface RefObject<T> {
  readonly current: T | null;
}
```

So even if we check that `tabGroupRef.current` exists before calling the function, we still have to use optional chaining in the function body.

But I'm fine with adding `&& tabGroupRef.current`

Reply:
We do? Can't we use the NonNullable magic?

Reply:
I suppose we can do something like that if we reach into the `ref` and mark `current` as non-nullable.

```ts
export function scrollTabContentIntoView(
  current: NonNullable<React.RefObject<HTMLElement>['current']>,
) {
  current.scrollIntoView({ behavior: 'smooth', block: 'start' })
}
```

And then

```ts
if (isMounted() && isScreenBelowLg && tabGroupRef.current) {
  scrollTabContentIntoView(tabGroupRef.current)
}
```

What do you think about that? I'm not used to doing it but it works :)

Reply:
Works for me! LGTM!